### PR TITLE
#164 Finalize environment for dynamic model use

### PR DESCRIPTION
**Pull request by lecourtoisflo**

### CODE REVIEW

File: scripts/envDFL.sh [hunk 1]
```diff
@@ -60,6 +60,34 @@ export_var_env_force() {
...
+    return
+  fi
+
+  if [ "$value" = UNDEFINED ]; then
+    error_exit "You must define the value of $name"
+  fi
+  export $name="$value"
+}
+
+export_var_env_dynawo() {
```
Comment (rosiereflo) [inline scripts/envDFL.sh:84]: why adding a method with the exact same objectives as  export_var_env_force_dynawo?

File: scripts/envDFL.sh [hunk 2]
```diff
@@ -125,18 +153,27 @@ set_environment() {
...
-    export DYNAWO_DDB_DIR=$DYNAWO_INSTALL_DIR/ddb
-    export DYNAWO_IIDM_EXTENSION=$DYNAWO_INSTALL_DIR/lib/libdynawo_DataInterfaceIIDMExtension.so
-    export DYNAWO_LIBIIDM_EXTENSIONS=$DYNAWO_INSTALL_DIR/lib
+
+    # dynawo vars that can be replaced by external
+    export_var_env_dynawo DYNAWO_DDB_DIR $DYNAFLOW_LAUNCHER_EXTERNAL_DDB $DYNAWO_INSTALL_DIR/ddb
+
+    # DYNAWO_IIDM_EXTENSION is used only in case of external IIDM extensions.
+    # Empty string is equivalent to not set for Dynawo
+    export DYNAWO_IIDM_EXTENSION=$DYNAFLOW_LAUNCHER_EXTERNAL_IIDM_EXTENSION
```
Comment (rosiereflo) [inline scripts/envDFL.sh:162]: Please use export_var_env_force

File: tests/main/reference/phase_shifter/outputIIDM.xml [hunk 3]
```diff
@@ -0,0 +1,127 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<iidm:network xmlns:iidm="http://www.powsybl.org/schema/iidm/1_4" id="ieee14bus" caseDate="2017-06-09T10:14:24.146000+02:00" forecastDistance="0" sourceFormat="CIM1">
```
Comment (rosiereflo) [inline tests/main/reference/phase_shifter/outputIIDM.xml:2]: What is the delta between this file and the input one?

**Review by rosiereflo — CHANGES_REQUESTED** [review]

File: scripts/envDFL.sh (hunk 2 above)
Comment (lecourtoisflo) [reply scripts/envDFL.sh:162]: OK

File: scripts/envDFL.sh (hunk 1 above)
Comment (lecourtoisflo) [reply scripts/envDFL.sh:84]: Renamed to export_var_env_dynawo_with_default

File: tests/main/reference/phase_shifter/outputIIDM.xml (hunk 3 above)
Comment (lecourtoisflo) [reply tests/main/reference/phase_shifter/outputIIDM.xml:2]: Delta on v and angles

**Review by rosiereflo — APPROVED** [review]

**Pull request merged**